skfp/utils/splitters.py

```diff
@@ -0,0 +1,239 @@
+from collections import defaultdict
+from collections.abc import Sequence
+from typing import Union
+
+import numpy as np
+from rdkit.Chem import Mol
+from rdkit.Chem.Scaffolds import MurckoScaffold
+
+from skfp.utils.validators import ensure_mols
+
+
+def scaffold_train_test_split(
+    data: Sequence[Union[str, Mol]],
+    train_size: float = 0.8,
+    test_size: float = 0.2,
+    return_indices: bool = False,
+    include_chirality: bool = False,
+) -> Union[
+    tuple[list[str], list[str]],
+    tuple[list[Mol], list[Mol]],
+    tuple[list[int], list[int]],
+]:
+    """
+    Split a list of SMILES or RDKit 'Mol' objects into train and test subsets using Bemis-Murcko [1]_ scaffolds.
+
+    This function ensures that similar molecules (sharing the same scaffold) are in the same split.
```

Comment:
Maybe focus more on the fact that train and test are structurally different? I mean, this is kind of the same, but the impact of the wording should stress that by using scaffold split, we strive for generalization on structurally novel molecules

Reply:
Also, MoleculeNet introduced scaffold split as an approximation to the time split, assuming that new molecules (test set) will be indeed structurally different in terms of scaffolds from the training set. Maybe mention this?